srv/components/system/config/hosts.sls

```diff
@@ -16,13 +16,15 @@
 #
 
 hostsfile:
-  file.managed:
+  file.blockreplace:
     - name: /etc/hosts
-    - contents: |
+    - backup: False
+    - marker_start: "#---pvt_data_start---"
```

Comment:
For unfamiliar reviewers:
These markers ensure that the changes are made only within this block.
If block is not found in the file, it would be appended to the end of file along with the markers.

Ref:
https://docs.saltstack.com/en/3001/ref/states/all/salt.states.file.html#salt.states.file.blockreplace